sql/parser/eval_test.go

```diff
@@ -308,3 +315,25 @@ func TestEvalExprError(t *testing.T) {
 		}
 	}
 }
+
+func TestSimilarEscape(t *testing.T) {
+	testData := []struct {
+		expr     string
+		expected string
+	}{
+		{`test`, `test`},
+		{`test%`, `test.*`},
+		{`_test_`, `.test.`},
+		{`_%*`, `..**`},
+		{`[_%]*`, `[_%]*`},
+		{`.^`, `\.\^`},
+		{`%(b|d)%`, `.*(?:b|d).*`},
+		{`se\"arch\"[\"]`, `se(arch)[\"]`},
```

Comment:
This was actually an artifact of `SimilarEscape`'s support for proper regexp `substring` escaping functionality. If you check out the substring section of the [PostgreSQL docs](http://www.postgresql.org/docs/9.0/static/functions-matching.html#FUNCTIONS-SIMILARTO-REGEXP), it discusses the use of an escape character followed by a double quote to mark the portion of the pattern to be returned. This was how the PostgreSQL `similar_escape` function was implemented, so I decided to include support here as well.